options/posix/include/sys/msg.h

```diff
@@ -2,13 +2,32 @@
 #define _SYS_MSG_H
 
 #include <sys/ipc.h>
+#include <bits/ansi/time_t.h>
+#include <abi-bits/pid_t.h>
 
 #ifdef __cplusplus
 extern "C" {
 #endif
 
+typedef unsigned long msglen_t;
+typedef unsigned long msgqnum_t;
+
 int msgget(key_t, int);
 
+struct msqid_ds {
+	struct ipc_perm msg_perm;
+	time_t msg_stime;
+	time_t msg_rtime;
+	time_t msg_ctime;
+	unsigned long __msg_cbytes;
+	msgqnum_t msg_qnum;
+	msglen_t msg_qbytes;
+	pid_t msg_lspid;
+	pid_t msg_lrpid;
+	unsigned long __glibc_reserved4;
+	unsigned long __glibc_reserved5;
```

Comment:
Is this in violation of the GPL?

Reply:
Can you copyright struct definitions? No clue, but if it's a problem, we can rename them.

Reply:
Well that's gonna be fun if I need struct definitions and musl doesn't have them. Can't use different names either as programs rely on them (and no, they aren't standard per se). I see no problems copying definitions, however, copying function implementations is indeed not done.

Reply:
Why are the reserved fields copied over anyways?

Reply:
Just rename them to `unsigned long __unused[2]` as musl does.